### PR TITLE
Stop eating spaces when a non-search widget is selected

### PR DESCRIPTION
**Pull request by marmarta**
fixes QubesOS/qubes-issues#8789

### CODE REVIEW

**Comment by marmarta** [comment]
codecov fails because unit tests don't cover sending key presses to menu

**Review by marmarek — CHANGES_REQUESTED** [review]
Now, when I press a space outside of the search page, keyboard navigation (arrows etc) stops working...

**Comment by marmarta** [comment]
:scream: I'll try to fix it

File: qubes_menu/appmenu.py [hunk 1]
```diff
@@ -252,13 +252,18 @@ def _key_press(self, _widget, event):
...
         if event.keyval == Gdk.KEY_Escape:
             self.hide_menu()
         if event.keyval == Gdk.KEY_space:
-            if not isinstance(self.get_active_window().get_focus(),
-                              Gtk.SearchEntry):
+            current_widget = self.get_active_window().get_focus()
+            if isinstance(current_widget,
+                          Gtk.SearchEntry):
+                p = current_widget.get_position()
+                current_widget.set_text(current_widget.get_text() + " ")
```
Comment (marmarek) [inline qubes_menu/appmenu.py:265]: Now when I type space in the middle of search entry (move cursor with arrows first), then the space doesn't appear there, only at the end...

**Review by marmarek — CHANGES_REQUESTED** [review]

File: qubes_menu/appmenu.py (hunk 1 above)
Comment (marmarta) [reply qubes_menu/appmenu.py]: FIXED IT FINALLY, no more raging consumption of spaces or weird space behavior

**Review by marmarek — APPROVED** [review]

File: qubes_menu/appmenu.py [hunk 2]
```diff
@@ -252,13 +252,18 @@ def _key_press(self, _widget, event):
...
         if event.keyval == Gdk.KEY_space:
-            if not isinstance(self.get_active_window().get_focus(),
-                              Gtk.SearchEntry):
+            current_widget = self.get_active_window().get_focus()
+            if isinstance(current_widget,
+                          Gtk.SearchEntry):
+                p = current_widget.get_position()
+                current_widget.insert_text(" ", p)
+                current_widget.set_position(p + 1)
                 return True
```
Comment (coyotebush) [inline qubes_menu/appmenu.py:267]: I suppose `current_widget` is the same as the widget receiving the event? If so, this return value (whether to stop propagation) seems backwards: the search widget should continue handling the event (by inserting a space itself), while the menu items should stop handling the event. The code before this change makes more sense to me.

**Pull request merged**